app/src/main/cpp/skyline/services/codec/IHardwareOpusDecoder.h

```diff
@@ -0,0 +1,110 @@
+     * @url https://switchbrew.org/wiki/Audio_services#IHardwareOpusDecoder
+     */
+    class IHardwareOpusDecoder : public BaseService {
+      private:
+        std::shared_ptr<kernel::type::KTransferMemory> workBuffer;
+        OpusDecoder *decoderState{};
+        i32 sampleRate;
+        i32 channelCount;
+        i32 decoderOutputBufferSize;
+
+        /**
+         * @brief Resets the codec state to be equivalent to a freshly initialized state
+         */
+        void ResetContext();
+
+        /**
+         * @brief Called by all DecodeInterleaved* functions
+         * @param performanceInfo Whether to return decode time taken or not
+         */
+        Result DecodeInterleavedImpl(ipc::IpcRequest &request, ipc::IpcResponse &response, bool performanceInfo = false);
+
+      public:
+        /**
+         * @param sampleRate The sample rate of the Opus audio data
+         * @param channelCount The channel count of the Opus audio data
+         */
+        IHardwareOpusDecoder(const DeviceState &state, ServiceManager &manager, i32 sampleRate, i32 channelCount, u32 workBufferSize, KHandle kWorkBuffer);
+
+        /**
+         * @brief Takes an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size and decoded sample count
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedOld
+         */
+        Result DecodeInterleavedOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Takes an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size, decoded sample count and decode time taken in microseconds
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedWithPerfOld
+         */
+        Result DecodeInterleavedWithPerfOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Same as DecodeInterleaved
+         * @see DecodeInterleaved
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleaved
+         */
+        Result DecodeInterleavedWithPerfAndResetOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Takes an input boolean flag, an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size, decoded sample count and decode time taken in microseconds
+         * @note The bool flag indicates whether or not a reset of the decoder context is being requested
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleaved
+         */
+        Result DecodeInterleaved(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
```

Comment:
I'd say make all of these descriptions more oriented towards function rather than purely input/output. The second line actually works fine it's just that the first line seems too much like a schema of inputs so really just chop off the first line for all of them.